test/integration/targets/mail/tasks/main.yml

```diff
@@ -0,0 +1,77 @@
+# TODO: Our current implementation does not handle SMTP authentication
+
+# NOTE: If the system does not support smtpd-tls (python 2.6 and older) we do basic tests
+- name: Attempt to install smtpd-tls
+  pip:
+    name: smtpd-tls
+    state: present
+  ignore_errors: yes
+  register: smtpd_tls
```

Comment:
Instead of ignoring errors here, I'd do a feature check for `ssl.create_default_context`. That will solve the issue of trying to use TLS when it's not available. It will also avoid accidentally skipping the TLS tests if the pip install fails due to network issues.

Reply:
That's true of dependencies for any test. Unless we have no other way to detect conditions which would prevent the installation, we shouldn't ignore the errors. Detect the condition(s) which we care about and skip the relevant portions of the test.

In this case relying on the pip install failure doesn't give us what we need (detecting lack of support for `create_default_context`). It also will cause the SSL tests to be skipped due to network failures, which we shouldn't be doing.

Reply:
The pip installation could fail because smtpd-tls is not supporting python 2.6, which is why I have that test and which is why I was ignoring the error. But even if it fails because of network-issues and it continues without TLS tests it is better than what we had before.

If you would like to improve it beyond this, feel free to do so.

Reply:
The pip install fails on Python 2.6 because it isn't supported. It succeeds on Python 2.7.6, which is also not supported. We need some way to skip testing with TLS in that case.

Make the pip install conditional on the availability of `ssl.create_default_context` and the tests will use TLS when it's available and skip the pip install when it isn't.

That will also eliminate the need for `ignore_errors: yes` on the install.

Reply:
I already spend too much work on this that should have gone elsewhere. So I fixed it quickly, feel free to improve it further. This is already a clear improvement over what we had before.

Reply:
Yes, having tests for this is a huge improvement. Thanks for putting this together.

It's a common theme putting together new tests. They look simple at first and then the corner cases take up 90% of the time required to finish them.